tests/testthat/helper.R

```diff
@@ -1,16 +1,40 @@
 Sys.setlocale("LC_COLLATE", "C") ## What CRAN does
 set.seed(666)
 
-## Our "test package" common harness code
-## `try` added because of some devtools::document weirdness
-try({
-    crunch_test_path <- system.file("crunch-test.R", package="crunch")
-    if (crunch_test_path == "") {
-        # hack for loadall
-        crunch_test_path <- "../../inst/crunch-test.R"
+# find a file starting with the current directory and looking in parent folders 
+# limited to levels_up for the file in file_name, additionally look inside the 
+# folder `inst` at each level looking for file_name as well.
+find_file <- function (file_name, levels_up = 3) {
+    for (i in 0:levels_up) {
+        pths <- file.path(".",
+                          paste0(rep("..", i), collapse = .Platform$file.sep),
+                          c("", "inst"), file_name)
+        fl <- pths[file.exists(pths)]
+        if (length(fl) == 1) return(fl)
+        if (length(fl)  > 1) {
+            warning("More than one file named ", file_name,
+                    " using the first one.")
+            return(fl[1])
+        }
     }
+    stop("Found no file named ", file_name, 
+         " Stopping tests because they will not function without ", file_name)
+}
+
+## Our "test package" common harness code
+crunch_test_path <- system.file("crunch-test.R", package="crunch")
+if (crunch_test_path == "") {
+    # hack for devtools::test / testthat::test_package
+    crunch_test_path <- find_file("crunch-test.R")
+}
+
+# only source crunch-test.R when the this is not being called (as the top most
+# call) by devtools::load_all or crunchdev::load_all
+loaded_by_load_all <- sys.calls()[[1]][[1]] == quote(devtools::load_all) |
```

Comment:
How about using `rstudioapi::isAvailable()`, like how we do in `login`: https://github.com/Crunch-io/rcrunch/blob/master/R/auth.R#L83-L84

Or even just `if (identical(.Platform$GUI, "RStudio"))`

Reply:
Both of those work for most situations, although they both prevent running `devtools::test()` interactively in RStudio. That's not something I do frequently (I'm much more likely to run them from RStudio's build pain or a new terminal in RStudio). The only time I can think of when running `devtools::test()` interactively is helpful is dropping a `browser()` in a problematic test.